### PR TITLE
Upgrade requirements and include yextend

### PR DESCRIPTION
**Pull request by austinbyers**
to: @jacknagz 
cc: @airbnb/binaryalert-maintainers 
size: medium
resolves: #76 
resolves: #77 
contributes to: #16 

## Background

In preparation for BinaryAlert v1.1 (which will add support for archive extraction), we need to upgrade all requirements and add [yextend](https://github.com/BayshoreNetworks/yextend) as a dependency.

## Changes

* Upgrade `terraform` from  ~0.10.4 to ~0.11.0
    * Change module output syntax if count = 0
    * Update documentation to reference TF 0.11 (instead of "latest version")
* Upgrade `terraform/aws-provider` from ~0.1.4 to ~1.3.1
* Upgrade `cbapi` from 1.3.2 to 1.3.4
* Upgrade `yara-python` from 3.6.3 to 3.7.0
* Add `yextend` 1.5.0 to the analyzer bundle (not used yet, but needed for archive analysis)
* Update READMEs about how to build native dependencies in Lambda AMI
* Fix bug: `terraform get` may be required before applying KMS ops in `./manage.py configure`
* Increase dispatcher timeout and dispatch limit (handles batch analysis much more efficiently)

## Testing

- CI: Unit tests, coverage, linting, static type enforcement
- `./manage.py configure` (no downloader)
- `./manage.py deploy`
- `./manage.py live_test`
- `./manage.py configure` (enable downloader)
- `./manage.py deploy`
- Live test of downloader

### CODE REVIEW

File: lambda_functions/analyzer/README.rst [hunk 1]
```diff
@@ -4,38 +4,61 @@ This Lambda function is the core of BinaryAlert. Each invocation downloads one o
...
+----------------------
+The YARA libaries used by BinaryAlert are natively compiled, and must therefore be pre-built on an
+Amazon Linux AMI in order to run in Lambda. This has already been done for you:
+``yara3.7.0_yextend1.5.zip`` contains the pre-built
+`yara_python <https://github.com/VirusTotal/yara-python>`_ (v3.7.0) and
+`yextend <https://github.com/BayshoreNetworks/yextend>`_ (v1.5) libraries for the Lambda environment
+and is automatically bundled with every deploy.
 
 If, however, you need to update or re-create the zipfile, SSH to an EC2 instance running the
 `AWS Lambda AMI <http://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html>`_
```
Comment (jacknagz) [inline lambda_functions/analyzer/README.rst]: have you considered automating this process with Packer or Chef?

File: lambda_functions/downloader/README.rst [hunk 2]
```diff
@@ -7,5 +7,21 @@ For more information, see the `documentation <https://binaryalert.io/uploading-f
 
 Cbapi Pip Dependency
 --------------------
-The ``cbapi`` library works best when pre-built on the Lambda AMI. Follow the same instructions given
-in the `analyzer README <../analyzer/README.rst>`_ to upgrade ``cbapi_1.3.2.zip`` when needed.
+The ``cbapi`` library needs to be pre-built on the AWS Lambda AMI. ``cbapi_1.3.4.zip`` is already
```
Comment (jacknagz) [inline lambda_functions/downloader/README.rst]: Packer with this too would be great

File: lambda_functions/analyzer/README.rst (hunk 1 above)
Comment (austinbyers) [reply lambda_functions/analyzer/README.rst]: @strcrzy and @ryandeivert have looked at this. Packer would be great, but we don't update the dependencies often enough to justify spending the time to automate it (yet)

Definitely something to consider! I might open an issue for it

File: lambda_functions/downloader/README.rst (hunk 2 above)
Comment (austinbyers) [reply lambda_functions/downloader/README.rst]: Yeah this is the same process as above

File: lambda_functions/analyzer/README.rst (hunk 1 above)
Comment (jacknagz) [reply lambda_functions/analyzer/README.rst]: that's fair, I only suggest because it's really easy to do

**Pull request merged**